### ISSUE TITLE
Time doesn't update along with publication date

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress#347]
Via @jorr

> I experience a very annoying behavior that once a publish date is
> set, actually *publishing *the post does't update the date to the
> current time. For some bizarre reason this happens regardless of whether I
> create the post through calendar (where I expect a publish date to be set,
> although I still expect publishing the post to update it) or through the WP
> Admin itself where I clearly don't expect a publishing date to be set. This
> kind of sounds like a *bug in PP *because I don't use the "Assigned"
> status as default for the Page post type and, expectedly, adding a new Page
> doesn't set a post date (it stays on "Publish immediately"). So there must
> be somewhere an action handler somewhere in your code that sets the default
> status from auto-draft to Assigned (or whatever the
> get_default_custom_status() returns) and also sets the post_date. This is
> probably unexpected behavior - it should only happen for posts created from
> Calendar in my opinion. But worse yet, publishing the post doesn't fix the
> date. There is a filter called *pp_calendar_allow_ajax_to_set_timestamp*
> but I wasn't able to understand how to use it to fix this.

**Comment by jorr** [comment]
@stevejburge @andergmartins any chance you guys had a look on this? :)

**Comment by andergmartins** [comment]
Hi @jorr please, could you clarify some points? 

- How are you publishing the post? Using quick edit? Bulk edit?
- Yes, the custom statuses can be set as default, but Pitch is what we enabled by default. What status you want as default? You can easily change this in the Statuses settings page. Hover a status and click on "Make default".

**Comment by jorr** [comment]
@andergmartins apologies for getting back late on this.

So here is the behavior I am seeing (for simplicity's sake I will just show the Calendar case):

1. Create a post from the Calendar on today's date.
2. Edit the post - notice the date is set on "Publish immediately".
3. Save the post - notice that after save the date is set on **today's date at 00:00**
4. Publish the post - notice that the date stays on **today's date at 00:00**

**Expected behavior**: the time should update to the time I clicked Publish.

A slightly alternative and even worse scenario:

1. Create a post from the Calendar on today's date.
2. Edit the post - notice the date is set on "Publish immediately".
3. Save the post - notice that after save the date is set on **today's date at 00:00**
4. Go back to Calendar and move the post one day back, then go to edit post again
5. Notice the publish date is now set to **yesterday's date at 00:00**
6. Publish the post - notice that the date stays on **yesterday's date at 00:00**

**Expected behavior: ** the time AND the date should update to the current datetime when I'm publishing the post.

In short - if the post is set for the future (w/r/t current datetime) then I get the "Schedule" button and all is cool. But if I publish the post on the same day, the post is published wrongly on 00:00 instead of the current time. And if I move the post around in the Calendar and don't publish it before the set date, then the date doesn't update and stays in the past, regardless of when I publish it.

Even shorter - Clickin publish should update the post datetime to the current moment.

**Comment by andergmartins** [comment]
Related to #546, fixed a few weeks ago.
